src/legacy/core_plugins/kibana/server/lib/management/saved_objects/relationships.js

```diff
@@ -17,50 +17,57 @@
  * under the License.
  */
 
+import { injectMetaAttributes } from './inject_meta_attributes';
+
 export async function findRelationships(type, id, options = {}) {
   const {
     size,
     savedObjectsClient,
     savedObjectTypes,
+    savedObjectsSchema,
   } = options;
 
   const { references = [] } = await savedObjectsClient.get(type, id);
 
-  // we filter the objects which we execute bulk requests for based on the saved
-  // object types as well, these are the only types we should be concerned with
-  const bulkGetOpts = references
-    .filter(({ type }) => savedObjectTypes.includes(type))
-    .map(ref => ({ id: ref.id, type: ref.type }));
+  // Use a map to avoid duplicates, it does happen but have a different "name" in the reference
+  const referencedToBulkGetOpts = new Map();
+  for (const { type, id } of references) {
+    referencedToBulkGetOpts.set(`${type}:${id}`, { id, type });
+  }
```

Comment:
optional: I've started to construct Map objects with the constructor which I find a bit more readable than populating it with the imperative API:

```js
  const referencedToBulkGetOpts = new Map(
    references.map(({ type, id }) => [`${type}:${id}`, { id, type }])
  );
```